apps/web/src/pages/Teams/TeamMembersPage/components/AddTeamMemberDialog.tsx

```diff
@@ -0,0 +1,203 @@
+    formState: { isSubmitting },
+  } = methods;
+
+  const handleSave = async (formValues: TeamMemberFormValues) => {
+    await executeMutation({
+      id: formValues.user,
+      user: {
+        roles: {
+          attach: {
+            roles: formValues.roles,
+            team: formValues.team,
+          },
+        },
+      },
+    })
+      .then((res) => {
+        if (!res.error) {
+          setIsOpen(false);
+          toast.success(
+            intl.formatMessage({
+              defaultMessage: "Member added successfully",
+              id: "AKQeBC",
+              description:
+                "Alert displayed to user when a team member's roles have been updated",
+            }),
+          );
+        }
+      })
+      .catch(() => {
+        toast.error(
+          intl.formatMessage({
+            defaultMessage: "Member role update failed",
+            id: "Ly2bBb",
+            description:
+              "Alert displayed to user when an error occurs while editing a team member's roles",
+          }),
+        );
+      });
+  };
+
+  const roleOptions = getTeamBasedRoleOptions(availableRoles, intl);
+  const userOptions = availableUsers.map((user) => ({
+    value: user.id,
+    label: getFullNameLabel(user.firstName, user.lastName, intl),
+  }));
+
+  const label = intl.formatMessage({
+    defaultMessage: "Add new member",
+    id: "+e2nr6",
+    description: "Label for the add member to team form",
+  });
+
+  return (
+    <Dialog.Root open={isOpen} onOpenChange={setIsOpen}>
+      <Dialog.Trigger>
+        <IconButton color="primary" mode="outline" icon={PlusIcon}>
+          {label}
+        </IconButton>
+      </Dialog.Trigger>
+      <Dialog.Content>
+        <Dialog.Header>{label}</Dialog.Header>
+        <Dialog.Body>
+          <p data-h2-margin="base(x1, 0)">
+            {intl.formatMessage({
+              defaultMessage:
+                "Search for a new member from the list of approved managers.",
+              id: "Mz61q/",
+              description:
+                "Help text for user field on the add member to team form",
+            })}
+          </p>
+          <FormProvider {...methods}>
+            <form onSubmit={handleSubmit(handleSave)}>
+              <Select
+                id="user"
+                name="user"
+                rules={{ required: intl.formatMessage(errorMessages.required) }}
+                label={intl.formatMessage({
+                  defaultMessage: "Manager name",
+                  id: "1BEtoY",
+                  description:
+                    "Label for the user select field on team membership form",
+                })}
+                options={userOptions}
+              />
+              {/** Note: Only one option since we are adding to this team's users */}
+              <input type="hidden" name="team" value={team.id} />
+              <Select
+                id="teamDisplay"
+                name="teamDisplay"
+                disabled
```

Comment:
Could we hide the "optional" decoration on this form element since it can't be interacted with?  Same for "manager" on the edit dialog.

Reply:
Good call, hidden in [hide optional label on disabled team input](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5811/commits/954afe5bf63d47fde577b18473e5038270ef39e8)